content/kapp-controller/docs/latest/packaging.md

```diff
@@ -186,3 +193,57 @@ status:
 **Note:** In this alpha release, values will only be included in the first
 templating step of the package, though we intend to improve this experience in
 later alpha releases.
+
+---
+## Artifact formats
+
+### Package bundle format
+
+Package bundle is an [imgpkg bundle](/imgpkg/docs/latest/resources/#bundle) that holds package contents such as Kubernetes YAML configuration, ytt templates, Helm templates, etc.
+
+Filesystem structure used for package bundle creation:
+
+```bash
+my-pkg/
+└── .imgpkg/
+    └── images.yml
+└── config/
+    └── deployment.yml
+    └── service.yml
+    └── ingress.yml
+```
+
+- `.imgpkg/` directory (required) is a standard directory for any imgpkg bundle
+  - `images.yml` file (required) contains container image refs used by configuration (typically generated with `kbld`)
+- `config/` directory (optional) should contain arbitrary package contents such as Kubernetes YAML configuration, ytt templates, Helm templates, etc.
+  - Recommendations:
+    - Group Kubernetes configuration into a single directory (`config/` is our recommendation for the name) so that it could be easily referenced in Package CR (e.g. using `ytt` template step against single directory)
+
+See [Creating a package](package-authoring.md#creating-a-package) for example creation steps.
+
+### Package Repository bundle format
+
+Package repository bundle is an [imgpkg bundle](/imgpkg/docs/latest/resources/#bundle) that holds Package CRs.
+
+Filesystem structure used for package repository bundle creation:
+
+```bash
+my-pkg-repo/
+└── .imgpkg/
+    └── images.yml
+└── packages/
+    └── simple-app.corp.com.1.0.0.yml
+    └── simple-app.corp.com.1.2.0.yml
+```
+
+- `.imgpkg/` directory (required) is a standard directory for any imgpkg bundle
+  - `images.yml` file (required) contains package bundle refs used by Package CRs (typically generated with `kbld`)
+- `packages/` directory (required) should contain zero or more YAML files describing available packages
```

Comment:
Do we want to say one or more YAML files here? What would be the use case for none?

Reply:
folder is required, but number of packages is actually 0+ in a repo. there might be cases when repo does actually include zero packages.

Reply:
Okay, sounds good